jdi-light-bootstrap-tests/src/test/java/io/github/epam/bootstrap/tests/composite/section/navbar/NavbarBrandTests.java

```diff
@@ -28,73 +27,69 @@ public void before() {
-                {"brand-as-image", ""},
-                {"brand-as-image-and-link", "Brand link"},
+                {navbarSection.navbarBrandLink, "Brand link"},
+                {navbarSection.navbarBrandHeading, "Brand heading"},
+                {navbarSection.navbarBrandAsImage, ""},
+                {navbarSection.navbarBrandAsImageAndLink, "Brand link"},
         };
     }
 
-    @Test
-    public void checkNavbarClickLink() {
-        for (int i = 1; i < navbarSection.allNavbarBrands.size() + 1; i++) {
-            UIElement nbb = navbarSection.allNavbarBrands.get(i);
-            if (nbb.hasAttribute("href")) {
-                nbb.highlight();
-                nbb.click();
-                WebDriver driver = WebDriverFactory.getDriver();
-                ArrayList<String> tabs = new ArrayList<>(
-                        WebDriverFactory.getDriver().getWindowHandles());
-                driver.switchTo().window(tabs.get(tabs.size() - 1));
-                assertEquals(getUrl(), navbarUrl);
-                driver.close();
-                driver.switchTo().window(tabs.get(tabs.size() - 2));
-            }
-        }
+    @DataProvider
+    public static Object[][] navbarBrandsWithImage() {
+        return new Object[][]{
+                {navbarSection.navbarBrandAsImage},
+                {navbarSection.navbarBrandAsImageAndLink},
+        };
     }
 
-    @Test(dataProvider = "navbarBrandData")
-    public void checkNavbarText(String navbarId, String navbarText) {
-        for (int i = 1; i < navbarSection.allNavbarBrands.size() + 1; i++) {
-            UIElement nbb = navbarSection.allNavbarBrands.get(i);
-            if (nbb.attr("id").equals(navbarId)) {
-                nbb.highlight();
-                nbb.is().core().text(navbarText);
-                nbb.unhighlight();
-            }
-        }
+    @DataProvider
+    public static Object[][] navbarBrandsWithLink() {
+        return new Object[][]{
+                {navbarSection.navbarBrandLink},
+                {navbarSection.navbarBrandAsImage},
+                {navbarSection.navbarBrandAsImageAndLink},
+        };
     }
 
-    @Test
-    public void checkNavbarClickImage() {
-        navbarSection.navbarBrandWithImage.stream()
-                .filter(nbb -> nbb.hasAttribute("href") && nbb.childs().size() > 0)
-                .map(nbbWithIm -> nbbWithIm.childs().get(1))
-                .forEach(imgFromNavbar -> {
-                    imgFromNavbar.highlight("blue");
-                    imgFromNavbar.is().attr("src", containsString(imgPath))
-                            .tag("img");
-                    imgFromNavbar.unhighlight();
+    @Test (dataProvider = "navbarBrandsWithLink")
+    public void  checkNavbarLink(UIElement brandAsLink) {
+        brandAsLink.is().core().hasAttr("href");
+        brandAsLink.highlight("blue");
+        brandAsLink.unhighlight();
+        brandAsLink.click();
+        int winNumber = WindowsManager.windowsCount();
```

Comment:
Method `WindowsManager.windowsCount()` return current count of browser tabs, it doesn't wait anything.  So, when you call it right after the click, result will be unpredictable. In such case it is better to write:

```
int winNumber = WindowsManager.windowsCount();
brandAsLink.click();
WindowsManager.switchToWindow(winNumber + 1);
```
Method `WindowsManager.switchToWindow(winNumber + 1)` will wait, while tab `winNumber + 1` will really apear.

Reply:
ok